apps/server/src/modules/authorization/README.md

```diff
@@ -0,0 +1,209 @@
+
+> This is not what we want in future.
+> For now, we must handle scope permissions, system user and user as seperated special cases and implementations.
+> By implementation of user permissions as school scope and others as instance scope we can do it in one way for all situations
+
+## How should you Authorised a Operation?
+
+The only point is in use cases (UC). They must solve the authorisation and orchestrate the logic that should be done in services, or private methods.
+You should never implement authorisations on service level, to avoid different authorisation steps.
+By calling other micro service for already authorised operations please use API_KEYS.
+
+> Not implemented but coming soon.
+
+## How to use Authorisation Service
+
+### Example 1 - ...
+
+{{ add more }}
+
+### Example 2 - ...
+
+{{ add more }}
+
+### Example 3 - ...
+
+{{ add more }}
+
+## Structure of the Authorisation Components
+
+### feathers-* (legacy/deprecated)
+
+It exist a adapter to call featherJS endpoints that solve authorisations.
+> This service is only used in news and should not use on any other place.
+> We want to remove it completly.
+
+### Authorisation Module
+
+The authorsation module is the core of authorisation. They collect all needed informations and handle it for you behind a small interface. It export the authoriation service that can be used in your use case over injections.
+
+### Reference.loader
+
+For situations where only the id and the domain object (string) type is knowen, it is possible to use the *ByReferences methodes.
+They load the reference directly.
+> Please keep in mind that it can be have impact of the performance if you use it wrongly.
+> We do it as seperate method to avoid using in areas where the domain object should be exist, but it is easyer to only pass the id.
+
+### shared/domain/rules/*
+
+The location to add new rules for entities / domain objects.
+
+#### permission-context.builder
+
+We export a permission context builder for prepare the parameter for the authorisation service called "authorisation context".
+This is optional and not required.
+But it enable us to change easy the structure of the authorisation context without touching many different places.
+
+### shared/domain/interface/*
+
+#### rolename.enum
+
+A enum that hold all avaible role names.
+
+#### permission.enum
+
+A enum that hold all avaible permission names. But mixing also all domain scopes atm.
+
+## Working other Internal MicroServices
+
+> Example FileService
+
+{{ add more }}
+
+## Legacy Tech Stack FeatherJS Hooks
+
+{{ add more }}
+
+## Desired Changes in Future
+
+Some small steps are done. But many open steps still exist.
+They follow our generell target
+
+### Next Steps
+
+1. implementation of Scope Based Permissions as generell solution instead of User Permissions that has only implicit school scopes for now
```

Comment:
```suggestion
1. Implementation of Scope Based Permissions as general solution instead of User Permissions that only have implicit school scopes for now
```